src/edu/washington/escience/myria/operator/CSVFileScanFragment.java

```diff
@@ -430,6 +441,23 @@ protected void init(final ImmutableMap<String, Object> execEnvVars) throws DbExc
                 new BufferedReader(new InputStreamReader(partitionInputStream)),
                 CSVFormat.newFormat(delimiter).withQuote(quote).withEscape(escape));
         iterator = parser.iterator();
+
+        try {
+          if (!iterator.hasNext()) {
```

Comment:
OK, but if you've read all the way to the end of the partition, couldn't you deduce that if `adjustedStartByteRange + byteOffsetFromTruncatedRowAtStart == partitionEndByteRange`, then there is nothing left to read? Alternatively, you could try to read one byte past the newline and set `flagAsIncomplete` if `read()` returned -1. I don't see why it's necessary to call `iterator.hasNext()` to determine this information.